beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -434,46 +418,24 @@ impl<T: BeaconChainTypes> DataAvailabilityCheckerInner<T> {
         f(self.critical.read().peek(block_root))
     }
 
-    fn block_import_requirement(
-        &self,
-        epoch: Epoch,
-    ) -> Result<BlockImportRequirement, AvailabilityCheckError> {
-        let peer_das_enabled = self.spec.is_peer_das_enabled_for_epoch(epoch);
-        if peer_das_enabled {
-            Ok(BlockImportRequirement::CustodyColumns(
-                self.custody_column_count,
-            ))
-        } else {
-            Ok(BlockImportRequirement::AllBlobs)
-        }
-    }
-
     /// Potentially trigger reconstruction if:
     /// - Our custody requirement is all columns
     /// - We >= 50% of columns, but not all columns
-    fn should_reconstruct(
-        &self,
-        block_import_requirement: &BlockImportRequirement,
-        pending_components: &PendingComponents<T::EthSpec>,
-    ) -> bool {
-        let BlockImportRequirement::CustodyColumns(num_expected_columns) = block_import_requirement
-        else {
-            return false;
-        };
-
+    fn should_reconstruct(&self, pending_components: &PendingComponents<T::EthSpec>) -> bool {
         let num_of_columns = self.spec.number_of_columns;
         let has_missing_columns = pending_components.verified_data_columns.len() < num_of_columns;
-
-        has_missing_columns
+        let is_super_node = self.custody_column_count == num_of_columns;
+        let is_reconstruction_possible =
+            pending_components.verified_data_columns.len() >= num_of_columns / 2;
+        is_super_node
+            && has_missing_columns
+            && is_reconstruction_possible
```

Comment:
Conflicts with 
- https://github.com/sigp/lighthouse/pull/6403

Reply:
yeah these two branches conflicts a lot and I'm planning to resolve them once one of them is merged. 

Another conflict is at the trigger point. They're both triggered when we have missing components - and we'd need to decide whether to fetch from EL or perform reconstruction in the same place. The benchmark for both are very similar but I think we should prioritise reconstruction if we have 50%, so we don't have to try fetch from EL, and fallback to reconstruction.

Reply:
Resolved conflicts in https://github.com/sigp/lighthouse/pull/6268/commits/d8eedf30c54c1a2cc0294d120717329b4371ebc7